src/portfft/descriptor.hpp

```diff
@@ -385,14 +390,14 @@ class committed_descriptor {
                             detail::level::SUBGROUP, static_cast<std::size_t>(factor_size), SubgroupSize,
                             {static_cast<Idx>(factor_sg), static_cast<Idx>(factor_wi)}, temp_num_sgs_in_wg) *
                         sizeof(Scalar) +
-                    2 * static_cast<std::size_t>(factor_size) * sizeof(Scalar)) <
+                    2 * static_cast<std::size_t>(factor_size) * (LocalRange / 2) * sizeof(Scalar)) <
```

Comment:
Was it for the  modifier the memory was not being accounted for ?

Reply:
Following our call to discuss this, I've put this PR back to draft status whilst I investigate. Call summary:
* The factor of 2 above may aready account for this. (I missed this factor of 2 in my accounting)
* This might cause the CI to pass just because it ups the local memory requirements a lot.

Reply:
I am not sure if this solution is correct, and I suspect this passes because it essentially would be forcing it use the workitem implementation only I suppose ..

I say so because - 
`num_scalars_in_local_mem`,  returns the memory in num_elements required for the input (which is padded). The store modifiers are also stored in local memory when using the subgroup implementation, and that too is padded. hence the amount of memory used by modifiers, and input would be the same, hence the result is multiplied by 2 in the lambda `fits_in_local_memory_subgroup`, when the input is of the layout BATCH_INTERLEAVED. When it's in the packed format, the store modifiers will not be used, and hence multiplication by 2 is omitted there. 

The line `2 * static_cast<std::size_t>(factor_size) * sizeof(Scalar))`  calculates the amount required for the twiddles of the subgroup implementation, and since they would be the same, for all the batches loaded, no need to multiply it by `LocalRange / 2`

Reply:
I've adjusted the calculation further in [8ecffbc](https://github.com/codeplaysoftware/portFFT/pull/127/commits/8ecffbce8282740f35d39be8d6a4d1c092c1533b).

A bug that may remain is that I'm not confident that the PACKED vs BATCH_INTERLEAVED is consistent internally between dispatch, and the detained global planning.